### ISSUE TITLE
creating a zonal cluster

### ISSUE THREAD

**Issue by Dramane-H** [terraform-google-modules/terraform-google-kubernetes-engine#1483]
### TL;DR

Hello guys,

I would like you to assist me in my task of creating a gke with a zonal cluster via your modules.

In fact, I am using your module to provision my GKE cluster, but i want it to be zonal not regional. 
It keeps saying this:
 Error: Cannot determine zone: set in this resource, or set provider-level zone.
│ 
│   with module.gke.data.google_container_engine_versions.region,
│   on .terraform/modules/gke/main.tf line 164, in data "google_container_engine_versions" "region":
│  164: data "google_container_engine_versions" "region" {

can someone provide me with some help?

### Terraform Configuration

```hcl
data "google_client_config" "default" {}

module "gke" {
  source                     = "terraform-google-modules/kubernetes-engine/google"
  project_id                 = var.project_id
  name                       = "${var.project_id}-platform-cluster"
  region                     = ""
  zones                      = var.zone
  network                    = google_compute_network.vpc.name
  subnetwork                 = google_compute_subnetwork.subnet.name
  ip_range_pods              = ""
  ip_range_services          = ""
  http_load_balancing        = false
  network_policy             = false
  horizontal_pod_autoscaling = true
  filestore_csi_driver       = false

  node_pools = [
    {
      name               = "${var.project_id}-platform-pool"
      machine_type       = "e2-medium"
      node_locations     = var.node_locations
      min_count          = 1
      max_count          = 3
      local_ssd_count    = 0
      spot               = false
      disk_size_gb       = 100
      disk_type          = "pd-standard"
      image_type         = "COS_CONTAINERD"
      enable_gcfs        = false
      enable_gvnic       = false
      auto_repair        = true
      auto_upgrade       = true
      service_account    = var.service_account
      preemptible        = false
      initial_node_count = 2
    },
  ]

  node_pools_oauth_scopes = {
    all = [
      "https://www.googleapis.com/auth/logging.write",
      "https://www.googleapis.com/auth/monitoring",
    ]
  }

  node_pools_labels = {
    all = {}

    default-node-pool = {
      default-node-pool = true
    }
  }

  node_pools_metadata = {
    all = {}

    default-node-pool = {
      node-pool-metadata-custom-value = "${var.project_id}-platform-pool"
    }
  }

  node_pools_taints = {
    all = []

    default-node-pool = [
      {
        key    = "default-node-pool"
        value  = true
        effect = "PREFER_NO_SCHEDULE"
      },
    ]
  }

  node_pools_tags = {
    all = []

    default-node-pool = [
      "default-node-pool",
    ]
  }

}
### Terraform Version

```sh
required_providers {
    google = {
      source  = "hashicorp/google"
      version = "4.44.1"
    }
  }
```

**Comment by bharathkkb** [comment]
Hi @Dramane-H , you can see an example here https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/34243391971fd539c5b4c5345fe74503162ec92d/examples/private_zonal_with_networking/main.tf#L66-L68. We use the region input to automatically find zones. Setting `regional` to false creates a zonal cluster. Feel free to reopen if still an issue.
